Documentation/High Availability and Scale.md

```diff
@@ -0,0 +1,447 @@
+    bits)
+
+2.  All connections/entries in the flow table are colored
+
+3.  Pairing relationship is established between two cards (primary card
+    and secondary card)
+
+4.  There exists a way to replicate connection (entry in a flow table)
+    to paired device.
+
+### "Perfect Sync" algorithm steps
+
+1.  **When card comes online, it chooses color** ("Color A")
+
+2.  As card starts receiving traffic it creates new flows (entries in
+    flow table)
+
+3.  New flows are added to the flow table with chosen color ("Color A")
+    and immediately replicated to paired device (colors do not get
+    replicated)
+
+4.  **All connections/flows** **use the same color until the pairing
+    fails**, and continues using the color if there is no pairing active
+
+5.  **When pairing is re-established, the device "changes" color to new
+    color ("Color B")**
+
+6.  New flows are added to the flow table using new color ("Color B")
+    and immediately replicated to paired device (colors do not get
+    replicated).
+
+7.  **Device starts replicating (sync) existing connections** (new
+    device to which either new pairing was created or existing device to
+    which pairing was re-established)
+
+8.  **Sync** method replicates **only** connections (entries in flow
+    table) which have color **different then actively used color**
+    (different then "Color B")
+
+9.  Entries with currently active color ("Color B") are not replicated
+    via sync algorithm, as they are replicated in real-time (immediately
+    as they are created).
+
+10. Algorithm finishes. Flow replication has been completed.
+
+### Notes
+
+-   The above algorithm ensures two sync are happening in parallel:
+
+    -   Real-time immediate sync of new changes (which will remain
+        active after "Perfect Sync" is completed)
+
+    -   Sync of existing connections (happening during "Perfect Sync")
+
+-   As the sync is happening, new connections as well as changes in
+    state of existing connections (irrespectively of color) is being
+    replicated immediately in real-time (outside of the sync algorithm).
+
+-   It is possible a connection will end (FIN) and will result in the
+    primary card removing flow and immediately sending the new
+    connection state change to close the connection to paired device
+    (before the existing connection was even journaled/synced to the
+    paired device). To deal with this possibility on the paired device
+    side, if the connection does not already exist in its table then
+    this update message should be ignored.
+
+-   When pairing is re-established, it is recommended that the secondary
+    card for simplicity to empty entire state of the flow table. This
+    will allow it to receive clean state.
+
+# References
+
+-   Visio Diagram -
+    <https://microsoft.sharepoint-df.com/teams/ProjectSirius/Shared%20Documents/General/Documentation/Design/MSFT%20Topology/AzureDedicated_Sirius_Diagrams.vsdx?web=1>
```

Comment:
These links aren't useful to the community, suggest you get the proper diagrams in the repo and point to them, or else put "TBD" or something.

Reply:
@chrispsommers  I was still working on the article. Now it should be ready.